contentcuration/contentcuration/view/node_views.py

```diff
@@ -0,0 +1,291 @@
+import copy
+import json
+import logging
+import os
+import uuid
+from django.http import HttpResponse, HttpResponseBadRequest
+from django.views.decorators.csrf import csrf_exempt
+from django.conf import settings
+from django.core.cache import cache
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import transaction
+from django.db.models import Q, Case, When, Value, IntegerField, Max, Sum
+from rest_framework.renderers import JSONRenderer
+from contentcuration.utils.files import duplicate_file
+from contentcuration.models import File, ContentNode, ContentTag, AssessmentItem
+from contentcuration.serializers import ContentNodeSerializer, SimplifiedContentNodeSerializer
+from le_utils.constants import format_presets, content_kinds, file_formats, licenses
+
+def get_total_size(request):
+    if request.method == 'POST':
+        data = json.loads(request.body)
+        sizes = ContentNode.objects.prefetch_related('assessment_items').prefetch_related('files').prefetch_related('children')\
+                    .filter(id__in=data).get_descendants(include_self=True)\
+                    .aggregate(resource_size=Sum('files__file_size'), assessment_size=Sum('assessment_items__files__file_size'))
+
+        return HttpResponse(json.dumps({'success':True, 'size': (sizes['resource_size'] or 0) + (sizes['assessment_size'] or 0)}))
+
+def delete_nodes(request):
+    if request.method == 'POST':
+        data = json.loads(request.body)
+        nodes = ContentNode.objects.filter(pk__in=data['nodes']).delete()
+        return HttpResponse({'success':True})
+
+def get_node_descendants(request):
+    if request.method == 'POST':
```

Comment:
This is called through the front-end. Basically, given a list of node ids (POST data), it would return the ids of the descendants of those nodes. The purpose of this is to calculate where a node can be moved to without putting it in the serializer for every node on the page